src/main/java/michaelscott/utils/Storage.java

```diff
@@ -0,0 +1,83 @@
+package michaelscott.utils;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import michaelscott.task.Task;
+import michaelscott.task.TaskList;
+import michaelscott.task.TaskParser;
+
+/**
+ * Handles storage of tasks, including reading and loading data from file
+ */
+public class Storage {
+    private final String filePath;
+
+    /**
+     * Constructs a storage class.
+     */
+    public Storage(TaskList taskList) {
+        try {
+            Files.createDirectories(Paths.get("./userdata"));
+        } catch (IOException e) {
+            Ui.showError(e.getMessage());
+        }
+        this.filePath = "./userdata/tasks.txt";
+        loadTasks(taskList);
+    }
+
+    public Storage() {
+        this.filePath = "./";
+    }
+
+    /**
+     * Reads the task from file and updates the task list.
+     *
+     * @param todo the task list.
+     */
+    public void saveTasks(ArrayList<Task> todo) throws MichaelScottException {
```

Comment:
maybe you can rename todo? As it may cause confusion with the Todo command u have